pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,429 @@
+            label. See :class:`TargetLabel` and :class:`~pyqtgraph.TextItem`
+        """
+        super().__init__(self)
         self.movable = movable
         self.moving = False
-        self.label = None
-        self.labelAngle = 0
-        self.pen = fn.mkPen(pen)
-        self.brush = fn.mkBrush(brush)
-
-    def setLabel(self, label):
-        if label is None:
-            if self.label is not None:
-                self.label.scene().removeItem(self.label)
-                self.label = None
+        self._label = None
+        self.mouseHovering = False
+
+        if pen is None:
+            pen = (255, 255, 0)
+        self.setPen(pen)
+
+        if hoverPen is None:
+            hoverPen = (255, 0, 255)
+        self.setHoverPen(hoverPen)
+
+        if brush is None:
+            brush = (0, 0, 255, 50)
+        self.setBrush(brush)
+
+        if hoverBrush is None:
+            hoverBrush = (0, 255, 255, 100)
+        self.setHoverBrush(hoverBrush)
+
+        self.currentPen = self.pen
+        self.currentBrush = self.brush
+
+        self._shape = None
+
+        self._pos = (0, 0)
+        if pos is None:
+            pos = (0, 0)
+        self.setPos(pos)
+
+        if isinstance(symbol, str):
+            try:
+                self._path = Symbols[symbol]
+            except KeyError:
+                raise KeyError("symbol name found in available Symbols")
+        elif isinstance(symbol, QtGui.QPainterPath):
+            self._path = symbol
+        else:
+            raise TypeError("Unknown type provided as symbol")
+
+        self.scale = size
+        self.setPath(self._path)
+        self.setLabel(label, labelOpts)
+
+    def setPos(self, pos):
+        """Method to set the position to ``(x, y)`` within the plot view
+
+        Parameters
+        ----------
+        pos : tuple, list, QPointF, QPoint, or pg.Point
+            Container that consists of ``(x, y)`` representation of where the
+            TargetItem should be placed
+
+        Raises
+        ------
+        TypeError
+            If the type of ``pos`` does not match the known types to extract
+            coordinate info from, a TypeError is raised
+        """
+        if isinstance(pos, tuple):
+            newPos = pos
+        elif isinstance(pos, list):
+            newPos = tuple(pos)
+        elif isinstance(pos, (QtCore.QPointF, QtCore.QPoint)):
+            newPos = (pos.x(), pos.y())
         else:
-            if self.label is None:
-                self.label = TextItem()
-                self.label.setParentItem(self)
-            self.label.setText(label)
-            self._updateLabel()
+            raise TypeError
+        if self._pos != newPos:
+            self._pos = newPos
+            super().setPos(Point(self._pos))
+            self.sigPositionChanged.emit(self)
 
-    def setLabelAngle(self, angle):
-        if self.labelAngle != angle:
-            self.labelAngle = angle
-            self._updateLabel()
```

Comment:
This method didn't get re-implemented in the new code. It might be enough to pass through `angle` to the `self._label.setAngle`, though the previous code did more with it than that, so I'm not sure.

Reply:
I should rephrase: I tried the code I proposed, and things _looked like_ they worked, so it wouldn't surprise me if that implementation is sufficient.

Reply:
The reason it works is my version I leverage the use of `TextItem.anchor`, so setting the angle rotates the label around the anchor point.  I think the original version didn't use `TextItem.anchor` and likely had to do more re-implementation to get the desired behavior that way.  (I'm going off of memory here, but I think that's why it "just works" without the use of this method).

Reply:
Fixed with 03394567e87d77e4f2308ce2e12aeac876db9fd8